cmd/webshorten/main.go

```diff
@@ -47,9 +51,23 @@ func shortenURL(w http.ResponseWriter, r *http.Request) {
 	w.Write(js)
 }
 
+var flagPort *int = flag.Int("port", 0, "use '-port=NUMBER' flag to run server on specific port(default: 5000)")
+const defaultPort string = ":5000"
+
+// ListenAndServeWrapper is a wrapper for http.ListenAndServ to let one specidfy PORT
+func ListenAndServeWrapper(a func(addr string, handler http.Handler) error) error {
```

Comment:
Using wrappers and passing functions as an argument is a bit hard to understand. :/ If you want to move arguments parsing to separate function you can use `func init()`. It is generally accepted way to move initialization code in package `main` that used by many gophers.
For example:
```golang
var port int

func init() {
	defPort := 5000
	if portVar, ok := os.LookupEnv("PORT"); ok {
		if portValue, err := strconv.Atoi(portVar); err == nil {
			defPort = portValue
		}
	}
	flag.IntVar(&port, "port", defPort, "port to listen on")
}
```